src/main/org/firebirdsql/gds/ng/wire/WireConnection.java

```diff
@@ -250,7 +275,7 @@ public final C identify() throws SQLException {
 
             xdrOut.writeInt(op_connect);
             xdrOut.writeInt(op_attach);
-            xdrOut.writeInt(CONNECT_VERSION2);
+            xdrOut.writeInt(CONNECT_VERSION3);
```

Comment:
Ok, so that would mean that the `getBytes()` could be changed to `getBytes(StandardCharsets.UTF_8)`. One thing I wonder is the use of UnixCrypt, because the crypt method seems to be applied directly to the first 8 char values (which in Java are UTF-16)